File: 01_Mean_Median_Mode/Solution.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution
{
    private List<int> BubbleSort(List<int> arr)
    {
        int n = arr.Count;
        for (int i = 0; i < n - 1; i++)
        {
            for (int j = 0; j < n - i - 1; j++)
            {
                if (arr[j] > arr[j + 1])
                {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
        return arr;
    }

    private int SumValue(List<int> ar)
    {
        return ar.Sum();
    }

    public double Mean(List<int> ar)
    {
        return SumValue(ar) / (double)ar.Count;
    }

    public double Median(List<int> ar)
    {
        ar = BubbleSort(ar);
        int n = ar.Count;
        if (n % 2 == 0)
        {
            return (ar[n / 2] + ar[n / 2 - 1]) / 2.0;
        }
        return ar[n / 2];
    }

    public (int, int) Mode(List<int> ar)
    {
        var frequency = new Dictionary<int, int>();
        foreach (var item in ar)
        {
            if (frequency.ContainsKey(item))
                frequency[item]++;
            else
                frequency[item] = 1;
        }

        int maxCount = frequency.Values.Max();
        int mode = frequency.FirstOrDefault(x => x.Value == maxCount).Key;
        return (mode, maxCount);
    }
}

File: 01_Mean_Median_Mode/Solution.py
class Solution:
    def bubble_sort(self, arr):
        n = len(arr)
        for i in range(n-1):
            for j in range(n-i-1):
                if arr[j] > arr[j+1]:
                    arr[j], arr[j+1] = arr[j+1], arr[j]
        return arr

    def sum_value(self, ar):
        return sum(ar)

    def mean(self, ar):
        return self.sum_value(ar) / len(ar)

    def median(self, ar):
        ar = self.bubble_sort(ar)
        n = len(ar)
        if n % 2 == 0:
            return (ar[n // 2] + ar[n // 2 - 1]) / 2
        return ar[n // 2]

    def mode(self, ar):
        frequency = {}
        for item in ar:
            frequency[item] = frequency.get(item, 0) + 1
        max_count = max(frequency.values())
        for key, value in frequency.items():
            if value == max_count:
                return key, max_count